osmdroid/src/main/java/org/odk/collect/osmdroid/OsmDroidMapFragment.java

```diff
@@ -82,21 +82,28 @@
  * A MapFragment drawn by OSMDroid.
  */
 public class OsmDroidMapFragment extends Fragment implements MapFragment,
-        MapEventsReceiver, IRegisterReceiver,
         LocationListener, LocationClient.LocationClientListener {
 
     // Bundle keys understood by applyConfig().
-    static final String KEY_WEB_MAP_SERVICE = "WEB_MAP_SERVICE";
-
-    @Inject
-    MapProvider mapProvider;
+    public static final String KEY_WEB_MAP_SERVICE = "WEB_MAP_SERVICE";
 
     @Inject
     ReferenceLayerRepository referenceLayerRepository;
 
     @Inject
     LocationClient locationClient;
 
+    @Inject
+    MapConfigurator mapConfigurator;
+
+    /**
+     * @deprecated {@link MapConfigurator} should encapsulate integration with
+     * {@link org.odk.collect.shared.settings.Settings}.
+     */
+    @Deprecated
```

Comment:
But it's needed in `MapFragmentDelegate` so should this be deprecated?

Reply:
I think, eventually, `MapConfigurator` could handle the `Settings` listener registration as well. I agree it might be a bit much to keep this deprecated in the code though - it's not an essential change.